static/js/github-create-branch.js

```diff
@@ -136,13 +136,23 @@ const createBranchPost = () => {
     })
     .fail((error) => {
       toggleSubmitDisabled(false);
-      if (error.responseJSON && error.responseJSON.err) {
-        showErrorMessage(error.responseJSON.err);
-      } else {
-        showErrorMessage("Please make sure all the details you entered are correct.")
+      switch (error.status) {
+        case 403:
+          // TODO: Fix the url later, once you figure out how to get the `installationId`
+          showValidationErrorMessage("s2id_ghRepo", "This GitHub repository hasn't been configured to your Jira site. <a href='#'>Allow access to this repository.</a>");
+          break;
+        case 422:
+          showValidationErrorMessage("branchNameText", "This GitHub branch already exists. Please use a different branch name.");
+          break;
+        case 404:
+          showValidationErrorMessage("s2id_ghParentBranch", "This GitHub source branch does not exist. Please use a different branch.");
+          break;
+        default:
```

Comment:
All above are server error messages which are returned by GitHub Api's.  It can't be  guaranteed that `404` only be thrown only in one case.  We should not display them as validation error messages.

Reply:
Removed the specific error message and listed as a list of possible causes.